### ISSUE TITLE
Allow no infusate

### ISSUE THREAD

**Issue by hepcat72** [Princeton-LSI-ResearchComputing/tracebase#1371]
### Feature Inspiration

Sometimes researchers analyze samples from animals that have no infusate (probably as a baseline/control?).  The infusate column is a required column in the `AnimalsLoader` and errors about the missing value prevent loading.

### Feature Description

Make the Infusate column in the `AnimalsLoader` optional and test the website to make sure that having no infusate doesn't cause problems.

### Alternatives

None

### Dependencies

None

### Comment

None

-----

### Assumptions

None

### Limitations

None

### Affected Components

None provided

### Requirements

None provided

#### Interface Change description

None provided

#### Code Change Description

None provided

#### Tests

None provided

**Comment by mneinast** [comment]
Yes this is a control for some recent experiments in the lab.

I'm not sure where tracebase gets the "labeled element" from.  Is it the Peak Annotation File?

**Comment by hepcat72** [comment]
I'm not entirely sure what is prompting your question, but where the labeled element comes from depends on the context. It can get it from the pecans file in the isotope label column, or it can get it from the tracers associated with the animal.

**Comment by hepcat72** [comment]
OK. Let me provide more context then. So what should happen, based on the current code, is, if there are labeled elements detected that don't exist in the tracers, you should get an unexpected label exception. It will be a warning that you can ignore, but it suggests the possibility that the sample could be contaminated or the tracers could be wrong. For the parent rows, you would just get "None" instead of a 0 count.

**Comment by mneinast** [comment]
> For the parent rows, you would just get "None" instead of a 0 count.

So, for example, would the PeakData for from one of these non-infused animals would look like...?

| Compound | Labeled Element:Count | Corrected Abundance |
|---|---|---|
| lactate | None | 54532  |
| lactate | C:1 | 0 |
| lactate | C:2 | 0 |
| lactate | C:3 | 0 |

Then the PeakGroup would be...?

| Compound | Labeled Element | Total Abundance |
|---|---|---|
| lactate | None | 54532  |

Or would Labeled Element be "C" in the PeakGroup table?

**Comment by hepcat72** [comment]
@mneinast - OK, so as I just mentioned in slack, I got the data to load for 1 peak annot file containing samples from an animal with no infusate.  The rendering of the advanced search table just has a bunch of empty cells.  Some have "None", but most are just blank.  Those are dead simple to fix.

There were some browsing pages with internal server errors.  I implemented a quick work-around, but there could exist problems elsewhere.  I suggest we discuss in the next meeting.  Or I can try and organize something earlier.  I know @jcmatese is wary about model changes this late before publication (as am I), and I think that concern is reasonably warranted.  I won't merge until we discuss.

I have a full load going in my sandbox that should finish tomorrow.  I think everything should go smoothly, but we'll see.
